File: programmers/77484.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

// 로또의 최고 순위와 최저 순위

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    sort(lottos.begin(), lottos.end());
    sort(win_nums.begin(), win_nums.end());

    int zero_count = 0;
    int is_same = 0;
    for (int i = 0; i < lottos.size(); i++) {
        if (lottos[i] == 0)
        {
            zero_count++;
            continue;
        }

        for (int j = 0; j < win_nums.size(); j++) {
            if (lottos[i] == win_nums[j])
            {
                is_same++;
                break;
            }
        }
    }

    int worst = is_same;
    int best = is_same + zero_count;

    if (is_same > 1)
        worst = 7 - is_same;
    else
        worst = 6;

    if (is_same + zero_count > 1)
        best = 7 - (is_same + zero_count);
    else
        best = 6;

    return { best, worst };
}

File: programmers/77484.py
## 로또의 최고 순위와 최저 순위

def solution(lottos, win_nums):
    unknown_count = 0
    hit_count = 0
    
    for lotto in lottos:
        if lotto == 0:
            unknown_count += 1
            continue
        
        if lotto in win_nums:
            hit_count += 1

    max_rank = 7 - (hit_count + unknown_count) if hit_count + unknown_count > 1 else 6
    min_rank = 7 - hit_count if hit_count > 1 else 6
    
    return [max_rank, min_rank]